components/dialog/dialog-confirm.js

```diff
@@ -85,13 +99,14 @@ class DialogConfirm extends DialogMixin(LitElement) {
 			</div>`;
 
 		const labelId = (this.titleText && this.text) ? this._titleId : this._textId;
+		const fullLabelId = this.critical ? `${this._offscreenId} ${labelId}` : labelId;
```

Comment:
Instead of hooking into `aria-labelledby` like this, could the `<d2l-offscreen>` stuff literally go _inside_ the `<h2>` above?

Reply:
It could, but the `h2` is only rendered if there's a title. We could do some conditionals where if there's a title then it goes there and if not then it goes with the text, that just gets a bit more complex but it's do-able.